### DISCUSSION TITLE
Combining Cloud-Carbon-Footprint with Keplar

### DISCUSSION THREAD

**Discussion by pygospa** [cloud-carbon-footprint/cloud-carbon-footprint#1329] [Q&A]
We've experimented with different tools, and found that a combination of CCF and [Kepler](https://github.com/sustainable-computing-io/kepler) might be very interesting: Keplar considers actual load from the pod by using the OSes tools to measure the load, and together with information on the hardware (power-consumptions) it calculates the power consumption in Watt and the CO2e when providing the electricity provider.

It is therefore much more accurate when it comes to measuring the power consumption of the running software, than CFF is, however it does not take in all the additional factors about the cloud provider (which CCF does).

Our idea would be to stripping the load assumptions from CCF and replacing them with Kepler measurements. Is this idea correct from your point of view, i.e. no part of the consumption is counted twice? And to remove the load assumptions from CCF: How exactly are they estimated? Can someone help this idea along in any way?

**Comment by ccasher** [comment]
Hi @pygospa, thanks for raising this issue! Regarding the question for how the load assumptions are estimated, we have our Compute estimation calculation stated [here](https://www.cloudcarbonfootprint.org/docs/methodology#compute). We assume an average utilization of 50% for compute usage which is explained in our documentation the trade-off for this decision. Please let us know if there is anything unclear or if you think there are any improvement to be made, even around making this more obvious for contributors!

We are not the most familiar with Kepler's offering, so it could be awesome if you provided any specific examples on how they are measure load with information on the hardware power-consumptions. If Kepler does give a more approximate average utilization figure, I think it would be totally reasonable to make this change in CCF and do away with our 50% average. Also in case you did not realize, we currently have an [integration underway with OpenCost](https://github.com/opencost/opencost/issues/2121) to combine methodologies. I am curious if you might think that the work here could be any way related to what you are proposing.

If you'd have any interest in opening a PR with some of the work you've done, please go ahead and we can continue to review it!
